src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +448,165 @@ impl ScalarField {
+                    // samples of mesh patches. The sample index specifies the
+                    // proper sample for the current voxel group.
+                    let mut sample_index = 0_u8;
+                    if corners_inside_volume_pattern[0] {
+                        sample_index |= 1;
+                    }
+                    if corners_inside_volume_pattern[1] {
+                        sample_index |= 2;
+                    }
+                    if corners_inside_volume_pattern[2] {
+                        sample_index |= 4;
+                    }
+                    if corners_inside_volume_pattern[3] {
+                        sample_index |= 8;
+                    }
+                    if corners_inside_volume_pattern[4] {
+                        sample_index |= 16;
+                    }
+                    if corners_inside_volume_pattern[5] {
+                        sample_index |= 32;
+                    }
+                    if corners_inside_volume_pattern[6] {
+                        sample_index |= 64;
+                    }
+                    if corners_inside_volume_pattern[7] {
+                        sample_index |= 128;
+                    }
+
+                    // The marching cubes lookup table sample, that contains
+                    // triplets of indices to the edge midpoints (currently
+                    // already vertices) that will become mesh patch faces.
+                    let marching_cubes_faces =
+                        marching_cubes_lookup_table[cast_usize(sample_index)].clone();
+
+                    // In some cases (when all the voxels are inside or outside
+                    // a volume), there is no mesh patch to be created.
+                    if !marching_cubes_faces.is_empty() {
+                        marching_cubes.push(
+                        // Generate new mesh path from all edge midpoint
+                        // vertices and only those faces that should be created
+                        // according th the marching cubes lookup table.
+                        Mesh::from_triangle_faces_with_vertices_and_computed_normals_remove_orphans(
+                            marching_cubes_faces,
+                            edge_midpoints,
+                            // Sharp is more efficient and gets lost by welding anyway
+                            NormalStrategy::Sharp,
+                        ),
+                    );
+                    }
+                }
+            }
+        }
+
+        // Join separate mesh planes into one mesh
+        let joined_voxel_mesh = tools::join_multiple_meshes(&marching_cubes);
+        let min_voxel_dimension = self
+            .voxel_dimensions
+            .x
+            .min(self.voxel_dimensions.y.min(self.voxel_dimensions.z));
+        // and weld naked edges.
+        tools::weld(&joined_voxel_mesh, (min_voxel_dimension as f32) / 4.0)
```

Comment:
Hmmm. I believe this should be possible to do without weld. Let's leave a fixme here to construct the mesh in-place.

Reply:
Again something I already gave a thought. Yes, it's possible, however you'd have to generate all the possible vertices and then remove the unused. Or you can create some sort of a "hash code" based on the vertex position, collect a list of such "hasehs", then calculate their cartesian position, then put them into some order and reindex the faces. Yes, that would be possible but also too much trouble, wouldn't it?

Reply:
> Yes, that would be possible but also too much trouble, wouldn't it?

That's why I only suggested to add the fixme.

Reply:
is this what you meant, though?